avidemux/qt4/ADM_userInterfaces/ADM_dialog/Q_encoding.cpp

```diff
@@ -440,6 +463,22 @@ void DIA_encodingQt4::keepOpen(void)
             QCoreApplication::processEvents();
         }
     }
+
+    if (deleteStats && outputFileName)
+    {
+        // try to delete stats files (they may not exists):
+        // filename_with_extension.stats
+        // filename_with_extension.stats.mbtree
+        // filename_with_extension.stats.cutree
+        char * tmpfn = (char*)ADM_alloc(strlen(outputFileName)+32);
+        #define DELETE_MACRO(x)	strcpy(tmpfn, outputFileName); strcat(tmpfn, x); \
+                ADM_info("Delete %s: %s\n",tmpfn,(remove(tmpfn)? "failed":"succeeded"));
+        DELETE_MACRO(".stats");
```

Comment:
I dont know, that it would be cleaner, while the mbtree and cutree are left as is.
The "mbtree" is defined in a patch, while "cutree" is nowhere to find.

Reply:
".mbtree" is appended by `int x264_ratecontrol_new( x264_t *h )` from encoder/ratecontrol.c to user-defined first pass log filename. The same applies to ".cutree" (coding unit tree) in `bool RateControl::init(const SPS& sps)` from x265 encoder/ratecontrol.cpp, i.e. we are fully in charge of the actual name of the first pass log file while different encoders mind their own business and take care of deleting existing stats files if necessary.

Reply:
I have no ratecontrol.cpp file...
Also searching for string "RateControl::init" give no results.

Reply:
This is in the source of the respective external libs we use.